diff -ru caml-shift/stacks.c caml-shift-modified/stacks.c
--- caml-shift/stacks.c	2018-02-26 06:31:12.000000000 +0100
+++ caml-shift-modified/stacks.c	2019-12-22 11:46:25.891863123 +0100
@@ -372,7 +372,7 @@
 value dbg_fatal_error(const value message)
 {
   myassert(Is_block(message) && Tag_val(message) == String_tag);
-  caml_fatal_error(String_val(message));
+  caml_fatal_error((char*)String_val(message));
   return Val_unit;		/* Doesn't return, actually */
 }
 
diff -ru caml-shift/stacks-native.c caml-shift-modified/stacks-native.c
--- caml-shift/stacks-native.c	2018-03-08 12:04:25.000000000 +0100
+++ caml-shift-modified/stacks-native.c	2019-12-31 20:07:01.194804976 +0100
@@ -143,6 +143,7 @@
 #include "mlvalues.h"
 #include "stack.h"
 #include "roots.h"
+#include "compatibility.h"
 #include <string.h>
 #include <custom.h>
 
@@ -985,7 +986,7 @@
 value dbg_fatal_error(const value message)
 {
   myassert(Is_block(message) && Tag_val(message) == String_tag);
-  caml_fatal_error(String_val(message));
+  caml_fatal_error((char*)String_val(message));
   return Val_unit;		/* Doesn't return, actually */
 }
 
