Author: Kate <kit.ty.kate@disroot.org>
Date:   Fri Jan 3 20:21:07 2020 +0000

    Add support for Coq 8.10 and 8.11

diff --git a/configure.in b/configure.in
index 25c731008..da1481d93 100644
--- a/configure.in
+++ b/configure.in
@@ -820,6 +820,12 @@ if test "$enable_coq_support" = yes; then
       8.9*)
          coq_compat_version="COQ89"
          ;;
+      8.10*)
+         coq_compat_version="COQ810"
+         ;;
+      8.11*)
+         coq_compat_version="COQ811"
+         ;;
       *)
          enable_coq_support=no
          AC_MSG_WARN(You need Coq 8.5 or later; Coq discarded)
diff --git a/opam/why3-coq.opam b/opam/why3-coq.opam
index b1e9340ff..899021451 100644
--- a/opam/why3-coq.opam
+++ b/opam/why3-coq.opam
@@ -39,7 +39,7 @@ depends: [
   "ocaml" {>= "4.05.0"}
   "ocamlfind" {build}
   "why3" {= version}
-  "coq" {>= "8.5" & < "8.10~"}
+  "coq" {>= "8.5" & < "8.12~"}
 ]
 
 depopts: [
